csharp/src/Ice/Connection.cs

```diff
@@ -574,10 +574,7 @@ public void Invoke()
                 {
                     Debug.Assert(Os != null);
                     Os.WriteSpan(Ice1Definitions.Magic.AsSpan());
-                    Os.WriteByte(Util.CurrentProtocol.Major);
-                    Os.WriteByte(Util.CurrentProtocol.Minor);
-                    Os.WriteByte(Util.CurrentProtocolEncoding.Major);
-                    Os.WriteByte(Util.CurrentProtocolEncoding.Minor);
+                    Os.WriteSpan(Ice1Definitions.PostMagic.AsSpan());
```

Comment:
We could also merge Magic/PostMagic if we want to consider the protocol/protocol encoding version as magic.

Reply:
I think `WriteSpan` + `AsSpan` is sufficient and correct. A span is very lightweight and "AsSpan" applies to many types.

The magic is `IceP`, not a longer byte string. `ProtocolAndProtocolEncodingVersion` emphasizes that the last 2 bytes correspond to the protocol encoding version (1.0) while we are actually using 1.1 since we're dropping 1.0 support. So maybe replace `PostMagic` by `ProtocolBytes`?

Reply:
`WriteMemory` could still be useful to users. It would avoid a function call here... not a big deal though since we're probably not after this kind of optimization so ok to keep AsSpan if you prefer it.

Reply:
Fixed with `ProtocolBytes`.